### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by jaywink**
* Add Python 3.6 Travis builds and update readme. Various tunage in the readme to make setting up environment clearer.
* Fix django-debug-toolbar for Django 1.10. Need to use an unreleased commit due to a bug in djdt related to django-jinja
* Fix requirements.apt for Ubuntu 16.04. As the current LTS, this should be the reference for development.
* Bump all the dependencies. Tests pass, so what could go wrong?
* Fix deprecation notice with pytest and setup.cfg

### CODE REVIEW

File: requirements/local.txt [hunk 1]
```diff
@@ -2,81 +2,84 @@
 # This file is autogenerated by pip-compile
 # To update, run:
 #
-#    pip-compile --output-file local.txt local.in
+#    pip-compile --output-file requirements/local.txt requirements/local.in
 #
-
-alabaster==0.7.8          # via sphinx
+-e git+https://github.com/jazzband/django-debug-toolbar.git@e174168683d9d81e918c2f5e4542079c7606f3ae#egg=django-debug-toolbar==1.6.0.1
```
Comment (charn) [inline requirements/local.txt:7]: I guess we don't need to install this dependency as an editable one i.e. can we lose the `-e`?

Comment (jaywink) [reply requirements/local.txt:7]: `pip-tools` only supports url based requirements if they have `-e` - tried losing it already :)

Comment (charn) [reply requirements/local.txt:7]: Well, I guess this will do then. :+1:

**Comment by charn** [comment]
LGTM! :shipit:

**Pull request merged**